language-plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -317,7 +347,48 @@ received evaluator
 --
 -- 1. 'EvaluationFailure' if at some point constants stop fitting into specified sizes.
 -- 2. A textual error if a PLC term can't be converted to a Haskell value of a specified type.
-type Convert a = ExceptT Text EvaluationResult a
+newtype ReflectT m a = ReflectT
+    { unReflectT :: ExceptT Text (InnerT EvaluationResult m) a
+    } deriving
+        ( Functor, Applicative, Monad
+        , MonadError Text
+        )
+
+-- GHC does not want to derive this for some reason.
+instance MonadTrans ReflectT where
```

Comment:
`deriving via ComposeT`? http://hackage.haskell.org/package/mmorph-1.1.2/docs/Control-Monad-Trans-Compose.html

Reply:
Indeed, I think `ComposeT` has most of the instances you want.

Reply:
It is somewhat different:

```
instance (MFunctor f, MonadTrans f, MonadTrans g) => MonadTrans (ComposeT f g)
  where
    lift = ComposeT . hoist lift . lift
```

I'm not sure why they first do the outer `lift` and then the inner one and not the other way around.

`deriving via` would not work for `Alternative`, because its logic is special and the instance has to be written by hand and can't be derived.

Reply:
Hm, is it different in a way that matters? If so, that makes me wonder if our version is wrong...

Reply:
> Hm, is it different in a way that matters?

I wouldn't expect it to be denotationally different, but it is slightly different operationally (which is not too important). Ok, I'll use `deriving via`, but mostly just to prove I'm not afraid of using it :)